src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java

```diff
@@ -940,8 +907,48 @@ public AssemblyResultSet generateEmptyLLocalAssemblyResult(final AssemblyRegion
         return resultSet;
     }
 
-    private static String dotFileName(final Haplotype refHaplotype, final int kmerSize, final String suffix) {
-        return (refHaplotype.getGenomeLocation() ==  null ? "" : IntervalUtils.locatableToString(refHaplotype) )
-                + "-sequenceGraph." + kmerSize + suffix;
+    /**
+     * Manage the printing of dot files and their naming convention, for example
+     * chr20:20-100-sequenceGraph.13.0.1.chain_pruned_read_threading_graph.dot
+     * chr20:20-100-sequenceGraph.13.0.2.cleaned_read_threading_graph.dot
+     * . . .
+     * chr20:20-100-sequenceGraph.13.1.0.non_ref_removed.dot,
+     * where 13 is the kmer size, and the following two numbers indicate broad categories (0 for the initial read threading
+     * de Bruijn graph, 1 for the later seq graph) and individual tags, respectively
+     * . . .
+     *
+     * This class automatically increments the section whenever a new graph is printed i.e. it switches from 0 to 1 once
+     * it is asked to print a transformation of the seq graph instead of the read threading graph
+     */
+    private class DotFilePrinter {
```

Comment:
This looks good. Can i ask you for one (very quick) test of this? can you run this over a small section of our test files with dbug graph printing enabled just to make sure that its correctly printing the dotfiles named correctly for each assembly region and graph within the assembly window (so regionsize.kmersize for both the 10 and 25mer graphs at most sites).

Reply:
or if you've already done a test like that (just make 100% sure its printing correctly across a whole region) then 👍

Reply:
I checked and it's working as before.